src/main/java/org/jabref/gui/mergeentries/MergeEntriesAction.java

```diff
@@ -52,7 +52,24 @@ public void execute() {
         BibEntry one = selectedEntries.get(0);
         BibEntry two = selectedEntries.get(1);
 
-        MergeEntriesDialog dlg = new MergeEntriesDialog(one, two);
+        // Compare the citation key
+        BibEntry first = one;
+        BibEntry second = two;
+
+        Optional<String> keyOne = one.getCitationKey();
+        Optional<String> keyTwo = two.getCitationKey();
+        if (keyOne.isPresent() && keyTwo.isPresent() && keyOne.get().compareTo(keyTwo.get()) > 0) {
+            first = two;
+            second = one;
+        }
+        else {
+            if (keyTwo.isPresent()) {
+                first = two;
+                second = one;
+            }
+        }
```

Comment:
There is org.jabref.logic.bibtex.comparator.EntryComparator:

    EntryComparator(false, true, InternalField.KEY_FIELD)

Thus:

```suggestion
        BibEntry first;
        BibEntry second;
        EntryComparator entryComparator = new EntryComparator(false, true, InternalField.KEY_FIELD);
        if (entryComparator.compare(one, two) <= 0) {
            first = one;
            second = two;
        } else {
            first = two;
            second = one;
        }
```

Since software engineering is an engineering prinicple and engineering includes testing, please add test cases. Add them to the class `org.jabref.logic.bibtex.comparator.EntryComparatorTest`